django/middleware/cache.py

```diff
@@ -121,6 +121,11 @@ def process_response(self, request, response):
             cache_key = learn_cache_key(
                 request, response, timeout, self.key_prefix, cache=self.cache
             )
+            # avoid multiple cache sets
+            if response.has_header("X-Cache-Success"):
+                return response
+            response["X-Cache-Success"] = "true"
```

Comment:
```suggestion
            # Avoid duplicating cache.
            response["X-Cache-Success"] = "true"
```
And have the if statement above be 
```python
        if (
            timeout
            and response.status_code == 200
            and not response.has_header("X-Cache-Success")
        ):
```

Reply:
We should not set a header that leaks back to the client!